src/app/clusters/test-cluster-server/test-cluster-server.cpp

```diff
@@ -197,122 +216,165 @@ CHIP_ERROR TestAttrAccess::WriteNullableStruct(AttributeValueDecoder & aDecoder)
+        gListOctetStringData[gListOctetStringDataLen].SetLength(entry.size());
+        gListOctetStringDataLen++;
+        return CHIP_NO_ERROR;
+    }
+    else
+    {
+        return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
     }
-
-    return iter.GetStatus();
 }
 
 CHIP_ERROR TestAttrAccess::ReadListLongOctetStringAttribute(AttributeValueEncoder & aEncoder)
 {
     // The ListOctetStringAttribute takes 512 bytes, and the whole attribute will exceed the IPv6 MTU, so we can test list chunking
     // feature with this attribute.
-    char buf[513] = "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef" // 5
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef" // 10
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef" // 15
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef" // 20
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef" // 25
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef"
-                    "0123456789abcdef" // 30
-                    "0123456789abcdef"
-                    "0123456789abcdef"; // 32 * 16 = 512
-    return aEncoder.EncodeList([buf](const auto & encoder) -> CHIP_ERROR {
-        for (uint8_t index = 0; index < kAttributeListLength; index++)
+    return aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
+        for (uint8_t index = 0; index < gListLongOctetStringLen; index++)
         {
-            ReturnErrorOnFailure(encoder.Encode(ByteSpan(chip::Uint8::from_const_char(buf), 512)));
+            ReturnErrorOnFailure(encoder.Encode(ByteSpan(chip::Uint8::from_const_char(sLongOctetStringBuf), 512)));
         }
         return CHIP_NO_ERROR;
     });
 }
 
+CHIP_ERROR TestAttrAccess::WriteListLongOctetStringAttribute(const ConcreteDataAttributePath & aPath,
+                                                             AttributeValueDecoder & aDecoder)
+{
+    if (!aPath.IsListItemOperation())
+    {
+        ListLongOctetString::TypeInfo::DecodableType list;
+
+        ReturnErrorOnFailure(aDecoder.Decode(list));
+
+        auto iter               = list.begin();
+        gListLongOctetStringLen = 0;
+        while (iter.Next())
+        {
+            const auto & entry = iter.GetValue();
```

Comment:
So this is just checking that all the values we are writing match sLongOctetStringBuf?  Should probably document that.